src/lib/permissions/hasAccessToSpace.ts

```diff
@@ -1,49 +1,47 @@
 import type { SpaceRole } from '@prisma/client';
 
-import type { SystemError } from '../../lib/errors';
-import {
-  InvalidInputError,
-  AdministratorOnlyError,
-  UserIsGuestError,
-  UserIsNotSpaceMemberError
-} from '../../lib/errors';
+import { InvalidInputError } from '../../lib/errors';
 import { prisma } from '../../prisma-client';
+import { isUUID } from '../utilities/strings';
+
+import type { PreComputedSpaceRole } from './core/interfaces';
 
 /**
  * @param userId - The ID of the user to check. If empty, the hasAccess should always return an error
  * @disallowGuest - If true, the user must be a member or admin of the space. If false, the user can be a guest
  */
-interface Input {
+type Input = {
   userId?: string;
   spaceId: string;
-  adminOnly?: boolean;
-  disallowGuest?: boolean;
-}
+} & PreComputedSpaceRole;
 
 interface Result {
-  error?: SystemError;
-  success?: boolean;
   isAdmin?: boolean;
-  spaceRole?: SpaceRole;
+  spaceRole: SpaceRole | null;
 }
 
-export async function hasAccessToSpace({ userId, spaceId, adminOnly = false, disallowGuest }: Input): Promise<Result> {
-  if (!spaceId || !userId) {
-    return { error: new InvalidInputError('User ID and space ID are required') };
+export async function hasAccessToSpace({ userId, spaceId, preComputedSpaceRole }: Input): Promise<Result> {
+  if (!spaceId || !isUUID(spaceId)) {
+    throw new InvalidInputError(`Valid space ID is required`);
+  } else if (
+    preComputedSpaceRole &&
+    (preComputedSpaceRole.userId !== userId || preComputedSpaceRole.spaceId !== spaceId)
+  ) {
+    throw new InvalidInputError(`SpaceRole userId and spaceId do not match the provided userId and spaceId`);
+  } else if (!userId) {
```

Comment:
why is userId allowed but spaceId is not? when would we call this without a user in context

Reply:
We don't always need a user ID. All permission compute stack has an optional userID as it may be called in the context where the user is an anon user interacting with public content

But we do need a space ID to evaluate correctly